semester_1/kr2.1/task2/graph/graph.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <stdlib.h>
+
+typedef struct Graph Graph;
+
+Graph *createGraph(const size_t nodeAmount, const size_t edgeAmount);
+void connectNode(Graph *graph, size_t index, size_t nodeNumber);
+void printComponentsTable(Graph *graph, const size_t nodeAmount);
+void deleteGraph(Graph * const graph);
```

Comment:
Лучше так:
```suggestion
void deleteGraph(Graph ** const graph);
```
Чтобы потом иметь возможность присвоить NULL в graph и не оставлять у вызывающего указатель на удалённую область памяти.